pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -503,6 +509,13 @@ func (a *ipAssigner) getAssignee(subnetInfo *crdv1b1.SubnetInfo, createIfNotExis
 	if err := util.EnsureRPFilterOnInterface(name, 2); err != nil {
 		return nil, err
 	}
+	// When the primary IP address is removed from the interface, promote a corresponding secondary IP address
+	// instead of removing all the corresponding secondary IP addresses. Otherwise, the deletion of one IP address
+	// can trigger the automatic removal of all other IP addresses in the same subnet, if the deleted IP happens to
+	// be the primary (first one assigned chronologically).
+	if err := util.EnsurePromoteSecondariesOnInterface(name); err != nil {
```

Comment:
@tnqn I added this here to match the call to `EnsureRPFilterOnInterface`, but it feels like we could move these calls to `addVLANAssignee`, so that they are called when the antrea-agent restarts. What do you think?

Reply:
Yes, it makes more sense to move them to `addVLANAssignee`, as there is no guarantee `EnsureRPFilterOnInterface` has executed once when the vlan interface is found.

Reply:
I will move both calls together in a follow-up PR.

Reply:
Follow-up PR: https://github.com/antrea-io/antrea/pull/6900